tools/tcat_ble_client/cli/base_commands.py

```diff
@@ -54,55 +55,112 @@ async def execute_default(self, args, context):
         if not response:
             return
         tlv_response = TLV.from_bytes(response)
         return CommandResultTLV(tlv_response)
 
 
-class CommissionCommand(Command):
+class HelloCommand(BleCommand):
+
+    def get_log_string(self) -> str:
+        return 'Sending hello world...'
+
+    def prepare_data(self, context):
+        return TLV(TcatTLVType.APPLICATION.value, bytes('Hello world!', 'ascii')).to_bytes()
 
     def get_help_string(self) -> str:
-        return 'Update the connected device with current dataset.'
+        return 'Send round trip "Hello world!" message.'
 
-    async def execute_default(self, args, context):
-        bless: BleStreamSecure = context['ble_sstream']
-        dataset: ThreadDataset = context['dataset']
 
-        print('Commissioning...')
+class CommissionCommand(BleCommand):
+
+    def get_log_string(self) -> str:
+        return 'Commissioning...'
+
+    def prepare_data(self, context):
+        dataset: ThreadDataset = context['dataset']
         dataset_bytes = dataset.to_bytes()
-        data = TLV(TcatTLVType.ACTIVE_DATASET.value, dataset_bytes).to_bytes()
-        response = await bless.send_with_resp(data)
-        if not response:
-            return
-        tlv_response = TLV.from_bytes(response)
-        return CommandResultTLV(tlv_response)
+        return TLV(TcatTLVType.ACTIVE_DATASET.value, dataset_bytes).to_bytes()
+
+    def get_help_string(self) -> str:
+        return 'Update the connected device with current dataset.'
+
 
+class DecommissionCommand(BleCommand):
 
-class DecommissionCommand(Command):
+    def get_log_string(self) -> str:
+        return 'Disabling Thread and decommissioning device...'
 
     def get_help_string(self) -> str:
         return 'Stop Thread interface and decommission device from current network.'
 
-    async def execute_default(self, args, context):
-        bless: BleStreamSecure = context['ble_sstream']
-        print('Disabling Thread and decommissioning device...')
-        data = (TLV(TcatTLVType.DECOMMISSION.value, bytes()).to_bytes())
-        response = await bless.send_with_resp(data)
-        if not response:
-            return
-        tlv_response = TLV.from_bytes(response)
-        return CommandResultTLV(tlv_response)
+    def prepare_data(self, context):
+        return TLV(TcatTLVType.DECOMMISSION.value, bytes()).to_bytes()
+
+
+class GetDeviceIdCommand(BleCommand):
+
+    def get_log_string(self) -> str:
+        return 'Retrieving device id.'
+
+    def get_help_string(self) -> str:
+        return 'Get unique identifier for the TCAT device.'
+
+    def prepare_data(self, context):
+        return TLV(TcatTLVType.GET_DEVICE_ID.value, bytes()).to_bytes()
+
+
+class GetExtPanIDCommand(BleCommand):
+
+    def get_log_string(self) -> str:
+        return 'Retrieving device id.'
+
+    def get_help_string(self) -> str:
+        return 'Get a URL for an application suited to commission device.'
```

Comment:
not sure if this is correct:
```suggestion
        return 'Get extended PAN ID that is commissioned in the active dataset.'
```